cylc/flow/prerequisite.py

```diff
@@ -223,6 +224,45 @@ def dump(self):
         # (Else trigger wiped out by pre-initial simplification.)
         return res
 
+    def api_dump(self, workflow_id):
+        """Return list of populated Protobuf data objects."""
+        if self.conditional_expression:
+            temp = self.get_raw_conditional_expression()
+            temp = temp.replace('|', ' | ')
+            temp = temp.replace('&', ' & ')
+        else:
+            for s_msg in self.satisfied:
+                temp = self.MESSAGE_TEMPLATE % s_msg
+        conds = []
+        num_length = int(math.ceil(float(len(self.satisfied)) / float(10)))
+        for ind, message_tuple in enumerate(sorted(self.satisfied)):
```

Comment:
Here too, I think we can use the enumerate + dictionary. Avoiding this way to access the dictionary and resolve `c_val = self.satisfied[message_tuple]` (i.e. one less array access operation).

```
for ind, (message_tuple, c_val) in enumerate(sorted(self.satisfied.items())):
```

Not tested, but I believe `sorted` will sort the `satisfied` dict per key, and using `.items()` will return the `key, value` pair.

Reply:
Fixed.